src/lib/krb5/os/changepw.c

```diff
@@ -143,6 +147,17 @@ kpasswd_sendto_msg_callback(SOCKET fd, void *data, krb5_data *message)
         local_kaddr.addrtype = ADDRTYPE_INET6;
         local_kaddr.length = sizeof(ss2sin6(&local_addr)->sin6_addr);
         local_kaddr.contents = (krb5_octet *) &ss2sin6(&local_addr)->sin6_addr;
+#ifndef _WIN32
+    } else if (local_addr.ss_family == AF_UNIX) {
+        /*
+         * There is no standard way to represent Unix domain sockets.  Assume
+         * that the receiver will accept a directional address if it verifies
+         * the source address.
+         */
+        local_kaddr.addrtype = ADDRTYPE_DIRECTIONAL;
```

Comment:
I wonder if we simply can copy `addr_directional_init` structure here? It would mean probably moving addr-utils around for linking purposes, though.

Reply:
The best alternative home for addr-utils.c is lib/krb5/os, with k5_ names and prototypes in k5-int.h.  (libkrb5support generally stays closer to the platform, so dealing in krb5_address structures would be an expansion of scope.)  changepw.c still couldn't use sockaddr_to_address() because it has the special requirement of maybe calling krb5_os_localaddr() (unless that's outdated, but we'd need to test on Windows to check).  It looks like print_addr() could possibly be leveraged in localaddr.c.

I previously landed on this placement and treating changepw.c as a special case, but I can't really articulate why I didn't like the libkrb5 private API placement as much.   I will try the experiment of moving it and see if I run into complications.